cli/src/pcluster/api/pcluster_api.py

```diff
@@ -388,21 +381,31 @@ def list_images(region: str):
             # get built images by image name tag
             images = AWSApi.instance().ec2.get_images()
             imagebuilders = [ImageBuilder(image=image, image_id=image.pcluster_image_id) for image in images]
-            images_response = [
-                ImageBuilderImageInfo(imagebuilder=imagebuilder, image=imagebuilder.image)
-                for imagebuilder in imagebuilders
-            ]
+            images_response = [ImageBuilderImageInfo(imagebuilder=imagebuilder) for imagebuilder in imagebuilders]
 
             # get building image stacks by image name tag
             imagebuilder_stacks = [
                 ImageBuilder(image_id=stack.get("StackName"), stack=ImageBuilderStack(stack))
                 for stack, _ in AWSApi.instance().cfn.get_imagebuilder_stacks()
             ]
             imagebuilder_stacks_response = [
-                ImageBuilderStackInfo(imagebuilder=imagebuilder, stack=imagebuilder.stack)
-                for imagebuilder in imagebuilder_stacks
+                ImageBuilderStackInfo(imagebuilder=imagebuilder) for imagebuilder in imagebuilder_stacks
             ]
 
             return images_response + imagebuilder_stacks_response
         except Exception as e:
             return ApiFailure(str(e))
+
+    @staticmethod
+    def _get_underlying_image_or_stack(imagebuilder: ImageBuilder, image_id: str):
```

Comment:
why passing `image_id` as parameter and not retrieve it from `imagebuilder.image_id`?

Reply:
done